### PR TITLE
Improve FinalizeInvocation logging

### PR DESCRIPTION
**Pull request by luluz66**
1. Add CtxInfo and CtxWarning
2. Add invocation_id to the context and use log.Ctx*** for logging.
3. Add additional logging after we successfully enqueue stats recorder task.

### CODE REVIEW

File: server/util/log/log.go [hunk 1]
```diff
@@ -302,6 +303,15 @@ func Infof(format string, args ...interface{}) {
 	log.Info().Msgf(format, args...)
 }
 
+// CtxInfo logs to the INFO log.
+// Logs are enriched with information from the context
+// (e.g. invocation_id, request_id)
+func CtxInfo(ctx context.Context, message string) {
```
Comment (vadimberezniker) [inline server/util/log/log.go:309]: Can you add the other funcs (CtxError, CtxDebug, etc) for completeness?

File: server/build_event_protocol/build_event_handler/build_event_handler.go [hunk 2]
```diff
@@ -623,6 +626,7 @@ func (e *EventChannel) FinalizeInvocation(iid string) error {
 	}
 
 	ctx, cancel := background.ExtendContextForFinalization(e.ctx, 10*time.Second)
+	ctx = log.EnrichContext(ctx, log.InvocationIDKey, iid)
```
Comment (vadimberezniker) [inline server/build_event_protocol/build_event_handler/build_event_handler.go:629]: Can you move this to OpenChannel, that way the invocation_id will be available through out the entire lifecycle.

**Review by vadimberezniker — CHANGES_REQUESTED** [review]

File: server/util/log/log.go (hunk 1 above)
Comment (luluz66) [reply server/util/log/log.go:318]: Done

File: server/build_event_protocol/build_event_handler/build_event_handler.go (hunk 2 above)
Comment (luluz66) [reply server/build_event_protocol/build_event_handler/build_event_handler.go]: Done

**Review by vadimberezniker — APPROVED** [review]

**Pull request merged**